### PR TITLE
fix(python): don't lose track of `ones` and `zeros` dtype, improve use with `Array`, raise error if dtype invalid

### PR DESCRIPTION
**Pull request by alexander-beedie**
Various minor fixes/improvements to the `ones` and `zeros` convenience functions.
(Follow up from https://github.com/pola-rs/polars/issues/13121#issuecomment-1864318096).

* Raise a suitable error on invalid/unsuitable "dtype" value instead of returning a series of `null` values.
* Ensure proper "dtype" pass-through from `repeat` → `parse_as_list_of_expressions` → `lit`.
* Extended test coverage to include error conditions, expected defaults, additional dtypes.
* Improve handling of `Array` dtype (get width and create 1/0 arrays of that length).

#### Improve handling of Array dtype:
```python
pl.ones(n=2, dtype=pl.Array(pl.UInt8,width=3), eager=True)
```
Before
```
shape: (2,)
Series: 'ones' [null]
[
    null
    null
]
```
After
```
shape: (2,)
Series: 'ones' [array[u8, 3]]
[
   [1, 1, 1]
   [1, 1, 1]
]
```
#### Raise on unsuitable/invalid dtype:
```python
pl.zeros(n=2, dtype=pl.Struct({"x": pl.Null, "y": pl.Duration}), eager=True) 
```
Before
```
shape: (2,)
Series: 'zeros' [struct[2]]
[
    {null,null}
    {null,null}
]
```
After
```
TypeError: invalid dtype for `zeros`; found Struct({'x': Null, 'y': Duration}))
```

### CODE REVIEW

**Comment by alexander-beedie** [comment]
(Rebased to pick up https://github.com/pola-rs/polars/pull/13327 fix for `Array` dtype).

File: py-polars/polars/functions/repeat.py [hunk 1]
```diff
@@ -112,13 +120,27 @@ def repeat(
...
+    value = parse_as_expression(value, str_as_lit=True, list_as_lit=True, dtype=dtype)
     expr = wrap_expr(plr.repeat(value, n._pyexpr, dtype))
     if eager:
         return F.select(expr).to_series()
     return expr
 
 
+# dtypes that have a reasonable one/zero mapping;
+# for anything more elaborate should use `repeat`
+_ones_zeros = {
```
Comment (ritchie46) [inline py-polars/polars/functions/repeat.py:132]: Can we put this in a RLU function?

Comment (alexander-beedie) [reply py-polars/polars/functions/repeat.py:132]: Ahh, good plan; no need to do unnecessary work on import👌 
Coming right up...

Comment (alexander-beedie) [reply py-polars/polars/functions/repeat.py]: Done

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**